console/v1/types_console_yaml_sample.go

```diff
@@ -27,6 +27,11 @@ type ConsoleYAMLSampleSpec struct {
 	Description ConsoleYAMLSampleDescription `json:"description"`
 	// yaml is the YAML sample to display.
 	YAML ConsoleYAMLSampleYAML `json:"yaml"`
+	// snippet indicates that the YAML sample is not the full YAML resource
```

Comment:
So this implies its a snippet that can be injected into whatever bit of yaml is already loaded into the window?  I assume if there is no yaml loaded it can't be injected?  would it overwrite existing bits of yaml if there is a collision?

Reply:
The snippets currently are coded to be a YAML insert rather than a YAML replace. So if you have "no content" (a blank YAML) the snippet would just insert the lines as the first set of content.

It does no validation when being inserted.

Reply:
I should note, it will replace any highlighted text as well. So if you select all your YAML and do an insert of a snippet - it'll effectively replace the content. Which likely is not the desire, since a snippet is unlikely to have the necessary top-level properties to work (but nothing stops it from doing so).

It's an open ended addition to the way Samples work. They exist as another tab in the UI so one can pick from samples or snippets independently.

Reply:
Makes sense.  Sounds like could be a bit gnarly usability at first, but fine once you get the hang of it.